client/src/components/Header/index.tsx

```diff
@@ -11,8 +14,72 @@ import AccountCircle from '@mui/icons-material/AccountCircle';
 import NotificationsNoneIcon from '@mui/icons-material/NotificationsNone';
 import MenuIcon from '@mui/icons-material/Menu';
 import logoSrc from '../../assets/logo.png';
+import { useNavigate } from 'react-router-dom';
+import { getProductsBySearch } from '@/services/Walmart';
+
+function useClickOutside(
+  ref: React.RefObject<HTMLElement>,
+  callback: () => void
+) {
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
+        callback();
+      }
+    }
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [ref, callback]);
+}
```

Comment:
nice to have a custom hook! could you move this under `utils/hooks/` or `hooks/`?